src/goCheck.ts

```diff
@@ -58,6 +60,19 @@ export function check(fileUri: vscode.Uri, goConfig: vscode.WorkspaceConfigurati
 	const runningToolsPromises = [];
 	const cwd = path.dirname(fileUri.fsPath);
 	const goRuntimePath = getBinPath('go');
+	const languageServerTool = getToolFromToolPath(getLanguageServerToolPath());
+	const languageServerOptions: any = goConfig.get('languageServerExperimentalFeatures');
+	let languageServerFlags: string[] = goConfig.get('languageServerFlags');
+	if (!Array.isArray(languageServerFlags)) {
+		languageServerFlags = [];
+	}
+
+	let disableBuild = languageServerOptions['diagnostics'] === true && (languageServerTool === 'gopls' || languageServerTool === 'bingo');
+	const disableVet = languageServerOptions['diagnostics'] === true && languageServerTool === 'gopls';
```

Comment:
can we put a comment on why we disable those when those language servers are used. ( i think its in the PR, but might get lost over time  )

Reply:
Done